Coupled-Cluster/spin_free_CC/EOM_CCSD.py

```diff
@@ -0,0 +1,190 @@
+import psi4
+import time
+import numpy as np
+np.set_printoptions(precision=5, linewidth=200, threshold=200, suppress=True)
+
+if __name__ == '__main__':
+    # EOM options
+    compare_psi4 = True
+    nroot = 3
+    nvec_per_root = 30
+    e_tol = 1.0e-6
+    max_iter = 80
+
+    # Psi4 setup
+    psi4.set_memory('2 GB')
+    psi4.core.set_output_file('output.dat', False)
+    # roots per irrep must be set to do the eom calculation with psi4
+    psi4.set_options({'basis': 'cc-pvdz', 'roots_per_irrep': [nroot]})
+    mol = psi4.geometry("""
+    O
+    H 1 1.1
+    H 1 1.1 2 104
+    noreorient
+    symmetry c1
+    """)
+
+    # Compute CCSD energy for required integrals and T-amplitudes
+    ccsd = helper_ccenergy(mol)
+    ccsd.compute_energy()
+
+    ccsd_cor_e = ccsd.ccsd_corr_e
+    ccsd_tot_e = ccsd.ccsd_e
+    print("\n")
+    print("{}{}".format('Final CCSD correlation energy:'.ljust(30, '.'),
+                        "{:16.15f}".format(ccsd_cor_e).rjust(20, '.')))
+    print("{}{}".format('Total CCSD energy:'.ljust(30, '.'),
+                        "{:16.15f}".format(ccsd_tot_e).rjust(20, ',')))
+
+    cchbar = helper_cchbar(ccsd)
+
+    # build CIS guess
+    ndocc = ccsd.ndocc
+    nvir = ccsd.nvirt
+    nov = ndocc * nvir
+    noovv = nov * nov
+    print("ndocc = {}".format(ndocc))
+    print("nvir = {}".format(nvir))
+    print("nov = {}".format(nov))
+
+    hbar_dim = nov + noovv
+    # L is the dimension of the guess space, we start with nroot*2 guesses
+    L = nroot * 2
+
+    # When L exceeds Lmax we will collapse the guess space so our sub-space
+    # diagonalization problem does not grow too large
+    Lmax = nroot * nvec_per_root
+
+    # An array to hold the excitation energies
+    theta = [0.0] * L
+
+    # build a helper_cceom object
+    cceom = helper_cceom(ccsd, cchbar)
+
+    # Get the approximate diagonal of Hbar
+    D = np.hstack((cceom.Dia.flatten(), cceom.Dijab.flatten()))
+
+    # We build a guess by selecting the lowest values of the approximate diagonal
+    # in the singles space one for each nroot*2 guesses we want,
+    # and we insert a guess vector with a 1 in the position corresponding to
+    # that single excitation and zeros everywhere else.
+    # This is a decent guess, more complicated one such as using CIS
+    # eigenvectors are more common in production level codes.
+    C_idx = D[:nov].argsort()[:nroot * 2]
+    C = np.eye(hbar_dim)[:, C_idx]
+
+    conv = False
+    eom_start = time.time()
+    for EOMCCSD_iter in range(0, max_iter + 1):
+        # QR decomposition ensures that the columns of C are orthogonal
+        C, R = np.linalg.qr(C)
+        L = C.shape[1]
+        theta_old = theta[:nroot]
+        print("EOMCCSD: Iter # {:>6} L = {}".format(EOMCCSD_iter, L))
+        # Build up the matrix S, holding the products Hbar*C, aka sigma vectors
+        S = np.zeros_like(C)
+        for i in range(L):
+            C1 = C[:nov, i].reshape(ndocc, nvir).copy()
```

Comment:
Might want to use a different variable than C it took me a second you were not unpacking the orbitals.